pkg/image/registryclient/client_mirrored.go

```diff
@@ -436,6 +468,16 @@ func (f blobMirroredBlobstore) Open(ctx context.Context, dgst digest.Digest) (di
 	err := f.repo.alternates(ctx, func(r RepositoryWithLocation) error {
 		var err error
 		rsc, err = r.Blobs(ctx).Open(ctx, dgst)
+		if err != nil {
+			klog.V(5).Infof("open %s from %s: %v", dgst, r.Named(), err)
+			return err
+		}
+
+		// Distribution's implementation of Open doesn't send any requests to
+		// the registry. We need the reader to send a request to see if the
+		// registry can serve the blob.
+		_, err = rsc.Read([]byte{})
```

Comment:
Yikes.  What are the performance characteristic changes to this?

This changes every trivial serve blob into a stat?

Reply:
In most cases it won't causes additional HTTP requests.

https://github.com/distribution/distribution/blob/1563384b69df9376389fe45ce949173a6383770a/registry/client/transport/http_reader.go#L66-L93

It forces the client to establish an HTTP connection and this connection will be reused by next reads. It won't be reused only if the caller decides to Seek before Read.

(Btw, we increased the number of requests by https://github.com/openshift/library-go/pull/1084, see https://github.com/openshift/image-registry/pull/278/commits/d7eeb86d675641dfcda4ba2dab0ef2f3d8e8e7b7; perhaps we need to check why)

Reply:
(`r.Blobs(ctx)` creates a BlobService with an internal cache, as we recreate BlobService on every call, it does not cache anything)

Reply:
(https://github.com/openshift/library-go/compare/master...dmage:cache-blobstore addresses the problem with the extra request, but as it is not related to the BZ, perhaps it should be a follow-up PR?)